sdks/python/apache_beam/io/gcp/bigquery_read_internal.py

```diff
@@ -207,6 +206,10 @@ def process(self,
           table_reference.datasetId,
           table_reference.tableId)
 
+      bq._delete_dataset(
```

Comment:
we should only delete the dataset if we were the ones creating it. Can you add logic to bigquery_tools around `create_temporary_dataset` that lets us know if we created the dtaset and thus should remove it?

Reply:
Sure, make sense. I'll work on it. Thanks @pabloem